nemo/collections/common/tokenizers/text_to_speech/tts_tokenizers.py

```diff
@@ -238,6 +238,71 @@ def __init__(
         )
 
 
+class RussianCharsTokenizer(BaseCharsTokenizer):
+    def __init__(
+        self,
+        punct=True,
+        apostrophe=True,
+        add_blank_at=None,
+        pad_with_space=False,
+        non_default_punct_list=None,
+        text_preprocessing_func=any_locale_text_preprocessing,
+    ):
+        """Russian phoneme-based tokenizer.
+        Args:
+            punct: Whether to reserve grapheme for basic punctuation or not.
+            apostrophe: Whether to use apostrophe or not.
+            add_blank_at: Add blank to labels in the specified order ("last") or after tokens (any non None),
+             if None then no blank in labels.
+            pad_with_space: Whether to pad text with spaces at the beginning and at the end or not.
+            non_default_punct_list: List of punctuation marks which will be used instead default.
+            text_preprocessing_func: Text preprocessing function for correct execution of the tokenizer. By default, it only applies lower() function.
+        """
+
+        ru_ipa = "əɪɐɨaʊioueʉæɵɛ" + "vrmtpsnʂkflzɫbdjɕxɡʐ" + "аеёиоуыэюя" + "бвгджзйклмнпрстфхцчшщъь"
```

Comment:
This assumes that every symbol in the strings is its own token. There are some caveats with how unicode handles characters with diacritics, so I would advise against this.

A better practice would be to have an honest list of tokens like in `collections/common/tokenizers/text_to_speechipa_lexicon.py`, e.g.
```
ru_tokens = ["ə", "ɪ", "ɐ", ...]
```

It is a bit unwieldy but will save debugging time down the road.

Reply:
@racoiaws , what do you think about the idea to create a subclass of BaseTokenizer that takes the alphabet from the config (like in ASR)? In this case **no need** to add dummy g2p classes and alphabets to the python code. It also allows to have models with slightly different alphabets for the same language. Wouldn't it be an easier way to add custom alphabets for other languages?

Reply:
Yes, that sounds great for experiments with pre-processed texts or external g2p.

It would need a list of symbols, a list of phonetic tokens, and some convention how to disambiguate between them in input strings (like phonetic symbols starting with # in IPATokenizer).

Note that the default NeMo way is still to do g2p on-the-fly, so when loading a pretrained model people could just run it on some text.